requirements/package.json

```diff
@@ -0,0 +1,7 @@
+{
+  "name": "torchgeo",
+  "private": "true",
+  "dependencies": {
+    "prettier": "^3.2.5"
```

Comment:
We should change in `.pre-commit-config.yaml` flag from `--write` to `--check` since it have version only up to `v3.1.0` because repo have been archived at 11th of April. But I suggest keeping it as a reminder for those who would forget to run `npx prettier . --write` before sending PR.

Reply:
3.2.5 is the newest version. I want to know what the oldest version that works for us is. For example, if our formatting relies on features introduced in 3.1, we would say `>=3.1` instead of `^3.2.5`. It's not super critical since it's only used in CI though.

Reply:
Yes and they are working already on 4.0.0 since last year so I assume it might come out in near future .Potentially we can use `>=3.0.0` because there are no breaking changes in 3.1.0 and 3.2.0 and as I checked release notes those small changes don't affect formats that we are using.

Reply:
There are no _documented_ breaking changes. That doesn't mean there aren't any breaking changes 😉 

It would be good to test with 3.0.0, 2.X, etc. and see which versions result in the same formatting. But this is optional.

Reply:
True, I will check are there any differences between 3.0.0 and 3.2.5. Below 3.0.0 I think that this would be problematic:
```md
Ignore .gitignored files by default ([#14731](https://github.com/prettier/prettier/pull/14731) by [@fisker](https://github.com/fisker))

Prettier ignores files ignored by .gitignore by default. If you want the old behavior(only ignore files ignored by .prettierignore), use

prettier . --write --ignore-path=.prettierignore
``` 
That was introduced in 3.0.0 and going below would cause formatting all env files IDE specific folders etc.

Reply:
I have just checked 3.0.0 vs 3.2.5 and after formatting files with 3.0.0 and then checking it with 3.2.5 all files remained unchanged, so we can use `>=3.0.0`. In that case I will leave `--write` flag in `.pre-commit-config.yaml` and update `package.json`.

There is also one thing, when we run `npm install requirements/` it creates `package.json` and lock in root directory so those two will also have to be added to `.gitignore` as `/*.json` to ignore them only in root.

Reply:
Could we run `npm install` from inside the `requirements` directory to prevent that? Is `npm install <path>` a common strategy? Maybe we need to move these config files to the root after all...

Reply:
Is it common I don't know since I haven't worked on many python/node projects, the one that I'm contributing to recently have them in root directory but it's a web app. At the moment I'm not on my working station so I can't check but I assume that running 'npm install' inside `requirements` folder would cause creating `node_modules` also in that directory but we could use `--prefix` for that. But also in that case this step need to be mentioned in `contributing.rst`.

I think that simplest solution would be best in that case it's either staying with `npm install requirements/` and `/*.json` in `.gitignore,` **or** moving `package.json` and lock too root and using `npm ci` instead.

Reply:
The files in `requirements` aren't used by contributors, they're only used in CI. The simplest solution is certainly to put everything in root, it just clutters root with stuff that isn't usually needed.

Reply:
But contributors will be installing it by them self and if they won't be instructed to do it the same way as in CI then again `package.json` will be created in root directory and either it will get send to PR or will need to be placed in `.gitignore`. So they would need to be instructed to install it the same way as it is in CI, other way such change wouldn't make sense.

That's why the most optimal solution is current one, where we ignore `/*.json` and install from `requirements/`.

Other solution is to remove part where contributor install and use `prettier` and obligate them to use `pre-commit`. But it's short term if we would like to use newer versions since hook for `prettier` is already archived and I don't know dose version 4.0.0 will have new hook.

Reply:
I don't see any issues with adding `node_modules` to `.gitignore`. It should ignore that directory regardless of if it's in the root or in `requirements/`.

Reply:
But not `node_modules` where an issue but adding `/*.json` to `.gitignore`.



> There is also one thing, when we run `npm install requirements/` it creates `package.json` and lock in root directory so those two will also have to be added to `.gitignore` as `/*.json` to ignore them only in root.

`node_modules` are already added since we discussed that.

Reply:
Ah sorry, I misunderstood. So if we replace `npm install requirements` with `npm install prettier` in the docs, we shouldn't have that issue, right?

Reply:
I dug some for and found a solution. We can use `npm install prettier --no-save` the `--no-save` flag will tell npm to not create `package.json` and lock files and will make a "clean" install. It would work with `npm install requirements/` as well.

Reply:
If it makes things easier, I'm not that opposed to keeping `package.json` in the root, although I would definitely prefer to keep the lock file in `requirements`.

Reply:
**Contributors:**
If we will set up `npm install prettier --no-save` or `npm install requirements/ --no-save` as default installation command we don't have issue with them creating `.json` file in root directory. It will just create `node_modules` and install latest `Prettier` version. So in `.gitignore` will be only included `node_modules`.

**CI:**
At the moment we use there `npm install requirements/` which reads from `.json` files we have set up there. Everything is working fine, CI commands are clean, we get to list dependencies and run check.
If we would like to use `npm ci` we would **have to** place both `.json` files in root directory since `npm ci` must have existing `package-lock.json` and `package-lock.json` need to have existing `package.json` to work.

Reply:
We might also try using `npm install prettier --no-save` in CI, if it will work there then we might not need `.json` files in `requirements/` either.

Reply:
We don't want this in CI. In CI, we want to enforce that a specific version of prettier is being installed, and we want dependabot to update this version in its own PRs. This prevents contributor PRs from failing due to new version updates that are unrelated to their PRs.

For contributors, I don't care too much either way. It's in `.gitignore`, so it shouldn't be an issue. But if you want to add that, I'm fine with adding that.